lib/core/src/Cardano/Pool/Performance.hs

```diff
@@ -103,25 +139,43 @@ apparentPerformance
         -- ^ Average performance
 apparentPerformance     [] = 0
 apparentPerformance epochs =
-    sum (n <$> epochs) / max 1 (sum (μ <$> epochs))
+    avg $ p <$> epochs
   where
-    n = double . poolProduction
-    μ e = _N * s / _S
+    p e = beta / sigma
       where
+        sigma = s / (max 1 _S) -- TODO: _S cannot be 0? throw?
+        beta = n / (max 1 _N)
+
         _N = double (totalProduction e)
         _S = double (totalStake e)
         s  = double (poolStake e)
+        n = double $ poolProduction e
+    avg xs = sum xs / (max 1 (fromIntegral (length xs)))
+
+
 
 -- | Read pool performances of many epochs from the database
 readPoolsPerformances
-    :: Monad m
+    :: MonadIO m
     => DBLayer m
     -> EpochNo
+    -> Tracer IO PerformanceLog
     -> m (Map PoolId Double)
-readPoolsPerformances DBLayer{..} currentEpoch = do
-    stats <- atomically $ forM historicalEpochs $ \ep -> mkEpochStats
-        <$> (count <$> readPoolProduction ep)
-        <*> (Map.fromList <$> readStakeDistribution ep)
+readPoolsPerformances DBLayer{..} currentEpoch tr = do
+    stats <- atomically $ forM historicalEpochs $ \ep -> do
+        prod <- count <$> readPoolProduction ep
+        stake <- Map.fromList <$> readStakeDistribution ep
+        let stats = mkEpochStats prod stake
+
+        -- FOR DEBUGGING
+        let p = PoolId $ unsafeFromHex "08b46d4bc5d8110c76dde2cc7a4c9dda0830c4434dedd7fcd32b1a4815c1a4f3"
+        case Map.lookup p stats of
```

Comment:
If we have ~1000 pools, that seems too big to just log in its entirety.

Improving logging in master might be tricky.

Reply:
maybe log only outliers? overperformers and underperformers by some margin. and also log newly discovered stake pools and say explicitly what performance they are attributed?